network.go

```
@@ -0,0 +1,765 @@
+	}
+
+	Ldebug.Printf("network info request: %s\n", jnetinfosReq)
+
+	// Setup the http request that will carry our json request
+	timeout := time.Duration(5 * time.Second)
+	client := http.Client{
+		Timeout: timeout,
+	}
+
+	request, err := http.NewRequest("POST", "https://"+networkCred.APIsrv+"/v1/netinfos", bytes.NewBuffer(jnetinfosReq))
+	if err != nil {
+		Lerror.Printf("failed to create a new https request: %v\n", err)
+		return
+	}
+	request.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(request)
+	if err != nil {
+		Lerror.Printf("failed network info request: %v\n", err)
+		return
+	}
+	defer resp.Body.Close()
+
+	// Read the response to our request
+	buf, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		Lerror.Printf("failed to parse network info response: %v\n", err)
+		return
+	}
+
+	Ldebug.Printf("netinfos: %v\n", string(buf))
+
+	err = json.Unmarshal(buf, &arrayCtrlInfo)
+	if err != nil {
+		Lerror.Fatalf("failed to unmarshal the controller information: %v\n", err)
+	}
+
+	if len(arrayCtrlInfo.NetInfos) <= 0 {
+		Lerror.Fatalf("failed to receive the controller information\n")
+	}
+
+	Ldebug.Printf("result: %v\n", arrayCtrlInfo)
+
+	Ldebug.Printf("ctrlInfo.Family: %s\n", arrayCtrlInfo.NetInfos[0].Family)
+	Ldebug.Printf("ctrlInfo.Addr: %s\n", arrayCtrlInfo.NetInfos[0].Addr)
+	Ldebug.Printf("ctrlInfo.Port: %s\n", arrayCtrlInfo.NetInfos[0].Port)
+	Ldebug.Printf("ctrlInfo.Region: %s\n", arrayCtrlInfo.NetInfos[0].Region)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	connController(ctx, cancel, &arrayCtrlInfo.NetInfos[0], &config)
+}
+
+func InitNetwork() {
+
+	// Setup ARP structs.
+	arpQueue = NewARPQueue(arpQueueMax)
+	arpTable = &ArpTable{}
+
+	gMAC = genMAC()
+
+	config := water.Config{
+		DeviceType: water.TUN,
+	}
+	config.Name = utunName
+
+	var err error
+	utun, err = water.New(config)
+	if err != nil {
+		Lerror.Fatalf("failed to initialize the %s interface: %v\n", utunName, err)
```

I would actually return an error and not run readUTun in main.go above if this fails to initialize

---

agreed, will update in the next PR